Modules/Project/Services/ProjectService.php

```diff
@@ -462,13 +462,17 @@ private function formatProjectFTEFOrExportAll($employees)
         return $teamMembers;
     }
 
-    public function getProjectsWithTeamMemberRequirementData()
+    public function getProjectsWithTeamMemberRequirementData($input)
     {
-        $projectsWithTeamMemberRequirement = Project::with('client')->status('active')
+        $query = Project::with('client')->status('active')
```

Comment:
Please update the all places. Please use the correct naming convention.
```suggestion
        $projects = Project::with('client')->status('active')
```

Reply:
SIr, I didn't name it '$projects' because the resultant is not projects data, rather the query is aggregated further and then after applying the get() function we get the the collection as named in the code. So do I still need to rename the variable?

Reply:
Can we write like this? Please test locally once and also please test all the edge cases. and check this query work as expected.
Please let me know, it works or not.

```
$projectsWithTeamMemberRequirement = Project::query()
    ->with('client')
    ->status('active')
    ->withCount('getTeamMembers as team_member_count')
    ->withSum('resourceRequirement as team_member_needed', 'total_requirement')
    ->havingRaw('team_member_needed - team_member_count')
    ->when($request, function ($query, $request) {
        return $query->where('name', $request);
    })
    ->get();
```

Reply:
Sir, it gives the result as expected and works fine. Thank You!